extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/intrumentation/vertx/RedisClientInstrumenterVertxTracer.java

```diff
@@ -0,0 +1,174 @@
+        InstrumenterBuilder<CommandTrace, Object> clientInstrumenterBuilder = Instrumenter.builder(
+                openTelemetry,
+                INSTRUMENTATION_NAME,
+                DbClientSpanNameExtractor.create(RedisClientAttributesGetter.INSTANCE));
+
+        this.redisClientInstrumenter = clientInstrumenterBuilder
+                .addAttributesExtractor(DbClientAttributesExtractor.create(RedisClientAttributesGetter.INSTANCE))
+                .addAttributesExtractor(RedisClientAttributesExtractor.INSTANCE)
+                .buildInstrumenter(SpanKindExtractor.alwaysClient());
+    }
+
+    @Override
+    public <R> boolean canHandle(R request, TagExtractor<R> tagExtractor) {
+        if (request instanceof CommandTrace) {
+            return true;
+        }
+
+        return "redis".equals(tagExtractor.extract(request).get("db.type"));
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <R> OpenTelemetryVertxTracer.SpanOperation sendRequest(
+            final Context context,
+            final SpanKind kind,
+            final TracingPolicy policy,
+            final R request,
+            final String operation,
+            final BiConsumer<String, String> headers,
+            final TagExtractor<R> tagExtractor) {
+        R commandTrace = (R) CommandTrace.commandTrace(tagExtractor.extract(request));
+        return InstrumenterVertxTracer.super.sendRequest(context, kind, policy, commandTrace, operation, headers, tagExtractor);
+    }
+
+    @Override
+    public <R> void receiveResponse(
+            final Context context,
+            final R response,
+            final OpenTelemetryVertxTracer.SpanOperation spanOperation,
+            final Throwable failure,
+            final TagExtractor<R> tagExtractor) {
+
+        InstrumenterVertxTracer.super.receiveResponse(context, response, spanOperation, failure, tagExtractor);
+    }
+
+    @Override
+    public Instrumenter<CommandTrace, Object> getReceiveRequestInstrumenter() {
+        return null;
+    }
+
+    @Override
+    public Instrumenter<CommandTrace, Object> getSendResponseInstrumenter() {
+        return null;
+    }
+
+    @Override
+    public Instrumenter<CommandTrace, Object> getSendRequestInstrumenter() {
+        return redisClientInstrumenter;
+    }
+
+    @Override
+    public Instrumenter<CommandTrace, Object> getReceiveResponseInstrumenter() {
+        return redisClientInstrumenter;
+    }
+
+    // From io.vertx.redis.client.impl.CommandReporter
+    static class CommandTrace {
+        private final Map<String, String> attributes;
+
+        CommandTrace(final Map<String, String> attributes) {
+            this.attributes = attributes;
+        }
+
+        static CommandTrace commandTrace(final Map<String, String> attributes) {
+            return new CommandTrace(attributes);
+        }
+
+        public String operation() {
+            return attributes.get("db.statement");
```

Comment:
The only thing is that those constants doesn't actually comes from OTEL, rather the Vert.x Command Reporter, and there it comes from the OpenTracing Semantic Conventions... `db.instance` is a no go for instance =]
I would rather have Vert.x enum being visible and use that as they are mapping it there instead of using OTEL conventions and it doesn't fully match... and if OTEL changes and vert.x doesn't we will have an issue...

Reply:
I see, can you please create constants?

Reply:
And make sure their values are being asserted in at least one test.

Reply:
created the constant, those values are indirectly tested by testing the span attributes, and all the attributes generated are being tested.

as those values are inlined by the compiler doesn't seem that constant are going to change much since this class is "hidden" as the vert.x one is also, as we don't use those values anywhere else =]